src/include/splash/basetypes/CollectionTypeGenerator.hpp

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright 2015 Carlchristian Eckert
+ *
+ * This file is part of libSplash.
+ *
+ * libSplash is free software: you can redistribute it and/or modify
+ * it under the terms of of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * libSplash is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with libSplash.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef COLTYPEGENERATOR_H
+#define COLTYPEGENERATOR_H
+
+
+#include "splash/basetypes/basetypes.hpp"
+#include "splash/CollectionType.hpp"
+
+namespace splash
+{
+
+#define COLLECTION_TEST(_name)                                                 \
+{                                                                              \
+    CollectionType* t = ColType##_name::genType(datatype_id);                  \
+    if(t != NULL && typeid(*t) == typeid(ColType##_name)) return t;            \
+}                                                                              \
+
+
+class CollectionTypeGenerator{
+public:
+
+    static CollectionType* genCollectionType(hid_t datatype_id)
```

Comment:
Consider making this a non-member function, as it does not seem to be related to CollectionTypeGenerator class.
Add a comment on the return, i.e. does the caller have to free/delete it?